Source/WebKit/GPUProcess/media/RemoteSourceBufferProxy.cpp

```diff
@@ -101,84 +101,74 @@ Ref<MediaPromise> RemoteSourceBufferProxy::sourceBufferPrivateDidReceiveInitiali
 
     segmentInfo.audioTracks = segment.audioTracks.map([&](auto& audioTrackInfo) {
         auto id = audioTrackInfo.track->id();
-        m_remoteMediaPlayerProxy->addRemoteAudioTrackProxy(*audioTrackInfo.track);
+        m_remoteMediaPlayerProxy->addRemoteAudioTrackProxy(Ref { *audioTrackInfo.track });
         m_mediaDescriptions.try_emplace(id, *audioTrackInfo.description);
         return InitializationSegmentInfo::TrackInformation { MediaDescriptionInfo(*audioTrackInfo.description), id };
     });
 
     segmentInfo.videoTracks = segment.videoTracks.map([&](auto& videoTrackInfo) {
         auto id = videoTrackInfo.track->id();
-        m_remoteMediaPlayerProxy->addRemoteVideoTrackProxy(*videoTrackInfo.track);
+        m_remoteMediaPlayerProxy->addRemoteVideoTrackProxy(Ref { *videoTrackInfo.track });
         m_mediaDescriptions.try_emplace(id, *videoTrackInfo.description);
         return InitializationSegmentInfo::TrackInformation { MediaDescriptionInfo(*videoTrackInfo.description), id };
     });
 
     segmentInfo.textTracks = segment.textTracks.map([&](auto& textTrackInfo) {
         auto id = textTrackInfo.track->id();
-        m_remoteMediaPlayerProxy->addRemoteTextTrackProxy(*textTrackInfo.track);
+        m_remoteMediaPlayerProxy->addRemoteTextTrackProxy(Ref { *textTrackInfo.track });
         m_mediaDescriptions.try_emplace(id, *textTrackInfo.description);
         return InitializationSegmentInfo::TrackInformation { MediaDescriptionInfo(*textTrackInfo.description), id };
     });
 
     // We need to wait for the CP's MediaPlayerRemote to have created all the tracks
-    return m_remoteMediaPlayerProxy->commitAllTransactions()->whenSettled(RunLoop::current(), [weakThis = ThreadSafeWeakPtr { *this }, this, segmentInfo = WTFMove(segmentInfo)](auto&& result) mutable -> Ref<MediaPromise> {
+    return m_remoteMediaPlayerProxy->commitAllTransactions()->whenSettled(Ref { RunLoop::current() }, [weakThis = ThreadSafeWeakPtr { *this }, this, segmentInfo = WTFMove(segmentInfo)](auto&& result) mutable -> Ref<MediaPromise> {
```

Comment:
We have RunLoop::protectedCurrent().

Reply:
Good point! I didn't realize this was there initially--thank you. As recommended above, I'm changing `current` to be `currentSingleton` so we should need this anymore.